# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.h
# Commit: c8bfef0a3dd3
# Full Hash: c8bfef0a3dd356d4ed84089b179684aa42d687ff
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-04-06 21:47:50
# Regressor Bug: 1562868
# File Overlap Count: 1
# Description:
#   Bug 1562868 - Limit the amount of data shared between processes when document.cookie is set, r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68179
# ==============================================================================

diff -r 68475f12d4f1 -r c8bfef0a3dd3 netwerk/cookie/CookieService.h
--- a/netwerk/cookie/CookieService.h	Mon Apr 06 13:09:19 2020 +0000
+++ b/netwerk/cookie/CookieService.h	Mon Apr 06 14:03:59 2020 +0000
@@ -31,10 +31,6 @@
 class CookiePersistentStorage;
 class CookiePrivateStorage;
 class CookieStorage;
-class CookieServiceParent;
-
-// these constants represent an operation being performed on cookies
-enum CookieOperation { OPERATION_READ, OPERATION_WRITE };
 
 // these constants represent a decision about a cookie based on user prefs.
 enum CookieStatus {
@@ -68,6 +64,8 @@
   NS_DECL_NSICOOKIEMANAGER
   NS_DECL_NSIMEMORYREPORTER
 
+  static already_AddRefed<CookieService> GetSingleton();
+
   CookieService();
   static already_AddRefed<nsICookieService> GetXPCOMSingleton();
   nsresult Init();
@@ -114,6 +112,11 @@
   nsresult Remove(const nsACString& aHost, const OriginAttributes& aAttrs,
                   const nsACString& aName, const nsACString& aPath);
 
+  bool SetCookiesFromIPC(const nsACString& aBaseDomain,
+                         const OriginAttributes& aAttrs, nsIURI* aHostURI,
+                         bool aFromHttp,
+                         const nsTArray<CookieStruct>& aCookies);
+
  protected:
   virtual ~CookieService();
 
@@ -170,8 +173,6 @@
                         const nsACString& aMaxage, int64_t aCurrentTime,
                         bool aFromHttp);
   void NotifyAccepted(nsIChannel* aChannel);
-  void NotifyRejected(nsIURI* aHostURI, nsIChannel* aChannel,
-                      uint32_t aRejectedReason, CookieOperation aOperation);
 
   nsresult GetCookiesWithOriginAttributes(
       const OriginAttributesPattern& aPattern, const nsCString& aBaseDomain,
@@ -202,13 +203,6 @@
   // private browsing.
   RefPtr<CookiePersistentStorage> mPersistentStorage;
   RefPtr<CookiePrivateStorage> mPrivateStorage;
-
-  // friends!
-  friend class DBListenerErrorHandler;
-  friend class CloseCookieDBListener;
-
-  static already_AddRefed<CookieService> GetSingleton();
-  friend class CookieServiceParent;
 };
 
 }  // namespace net